Map/LightOverlay.gd

```diff
@@ -1,106 +1,114 @@
+	map_upper_half.append(Vector2(room_container.get_size().x, first_room_rect.position.y - offset_position_y))
+	map_upper_half.append(Vector2(room_container.get_size().x, 0))
 	
-	floor_rect_packed_array_bottom_half.append(Vector2(0, first_room_rect.position.y - offset_position_y))
-	floor_rect_packed_array_bottom_half.append(Vector2(0, room_container.get_size().y))
-	floor_rect_packed_array_bottom_half.append(Vector2(room_container.get_size().x, room_container.get_size().y))
-	floor_rect_packed_array_bottom_half.append(Vector2(room_container.get_size().x, first_room_rect.position.y - offset_position_y))
+	map_bottom_half.append(Vector2(0, first_room_rect.position.y - offset_position_y))
+	map_bottom_half.append(Vector2(0, room_container.get_size().y))
+	map_bottom_half.append(Vector2(room_container.get_size().x, room_container.get_size().y))
+	map_bottom_half.append(Vector2(room_container.get_size().x, first_room_rect.position.y - offset_position_y))
 	
-	# If we got room circles to draw, then draw em. This will not draw anything if we don't have a torch or a player position put down anywhere.
+	# If we got room circles to draw, then draw em. 
+	# This will not draw anything if we don't have a torch or a player position put down anywhere.
 	if room_circles.size() > 0:
-		# Create the polygon of all of the lit room circles. 
+		# Create the outer polygon of all of the lit room circles.
+		# Get the first element of the lit room circles, 
+		# then iterate through the array, getting the next room circle and merging the polygons togetgher to get one polygon,
+		# while assigning it to the current_polygon.
 		var current_lit_room_polygon: PackedVector2Array = room_circles[0]
-		for room_circle_index in range(room_circles.size()):
+		for room_circle_index: int in range(room_circles.size()):
 			if (room_circle_index + 1 <= room_circles.size() - 1):
 				var merged_result = Geometry2D.merge_polygons(current_lit_room_polygon, room_circles[room_circle_index + 1])
 				current_lit_room_polygon = merged_result[0]
-
 		draw_polygon(current_lit_room_polygon, [Color(0, 0, 0, 0.25)])
-		var outer_polygons_top_half = Geometry2D.clip_polygons(floor_rect_packed_array_top_half, current_lit_room_polygon)
-		for outer in outer_polygons_top_half:
+		
+		# From the finished merged polygon blob from the lit room circles we created above, 
+		# Get the polygon inverse of the black overlay with the merged polygon blob we created above then draw them out.
+		var outer_polygons_upper_half = Geometry2D.clip_polygons(map_upper_half, current_lit_room_polygon)
+		for outer: PackedVector2Array in outer_polygons_upper_half:
 			draw_polygon(outer, [Color(0, 0, 0, 1)])
-		var outer_polygons_bottom_half = Geometry2D.clip_polygons(floor_rect_packed_array_bottom_half, current_lit_room_polygon)
-		for outer in outer_polygons_bottom_half:
+		var outer_polygons_bottom_half = Geometry2D.clip_polygons(map_bottom_half, current_lit_room_polygon)
```

Comment:
missing typing
You should put the untyped var as a warning in the debug options (advanced parameters, debug, GDScript)
It's done on the main but we don't have it here yet, so that would probably help for the time being